src/components/generic/asynchronous-button.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from "react";
+
+import { Button, ButtonProps } from "semantic-ui-react";
+
+interface Props extends Omit<ButtonProps, "onClick"> {
+  onClick?: () => void | Promise<void>;
+}
+
+/**
+ * Component state
+ */
+interface State {
+  loading: boolean;
+}
+
+/**
+ * Class for button component that performs asynchronous function calls
+ */
+class AsyncButton extends React.Component<Props, State> {
+
+  /**
+   * Component constructor
+   *
+   * @param props props
+   */
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      loading: false
+    }
+  }
+
+  /**
+   * Component render
+   */
+  public render = () => {
+    const { children, disabled } = this.props;
+    const { loading } = this.state
+
+    return (
+      <Button { ...this.props } disabled={ loading ?? disabled } loading={ loading } onClick={ this.handleClick }>
+        { children }
+      </Button>
+    );
+  }
+
+  /**
+   * Method for handling button press
+   */
+  private handleClick = () => {
+    this.handleAsync();
+  }
+
+  private handleAsync = async () => {
```

Comment:
Check if these can be combined to just one async handleClick method, button component should accept that

Reply:
You are right. This was probably only necessary on the app side with react-native buttons. Fixed to just one function.

Reply:
I was wrong. It isn't required in there either